jasmine/private/runner.js

```diff
@@ -1,4 +1,65 @@
 const path = require("path");
-const jasmine_path = require.resolve("jasmine");
 
-require(path.join(jasmine_path, "..", "..", "bin", "jasmine.js"));
+const shardCount = Number(process.env.TEST_TOTAL_SHARDS);
+const shardIndex = Number(process.env.TEST_SHARD_INDEX);
+const shardStatusFile = process.env.TEST_SHARD_STATUS_FILE;
```

Comment:
Interesting... 🤔 

I've been working with sharding via rules_nodejs `nodejs_test` for a while now and never seen this. Do you have more info other then the https://bazel.build/reference/test-encyclopedia#test-sharding ?

Reply:
This goes by round-robin partitioning in SQL world but a slightly different version of it. I found this https://www.researchgate.net/figure/The-four-partitioning-strategies-a-round-robin-b-range-c-list-and-d-hash_fig1_276075813

Reply:
Though it took the code from rules_jest.

Reply:
Sorry I specially mean the `TEST_SHARD_STATUS_FILE` status file...

Reply:
If you don't touch the file, Bazel will think the test runner doesn't support sharping

Reply:
Yeah that's what the docs say (the link above), but that's all it says.

Doing `nodejs_test(shard_count=N)` seems to work fine without touching this file. Does `nodejs_test` do something I'm not seeing?

With `nodejs_test` I'm consuming the 2 sharding variables and do the sharding very similar to this PR and it seems to be working fine. It always launches with N processes, the processes divide the specs manually etc. I guess I should try this and see if it changes anything...? 🤔